File: application/util/tator_localization_type.py
from enum import IntEnum


class TatorLocalizationType(IntEnum):
    BOX = 48
    DOT = 49
    SUB_BOX = 794
    SUB_DOT = 795

    @classmethod
    def is_relevant(cls, localization_type: int) -> bool:
        return localization_type in cls._value2member_map_

    @classmethod
    def is_box(cls, localization_type: int) -> bool:
        return localization_type == cls.BOX or localization_type == cls.SUB_BOX

    @classmethod
    def is_dot(cls, localization_type: int) -> bool:
        return localization_type == cls.DOT or localization_type == cls.SUB_DOT

File: application/static/js/util/tatorLocalizationType.js
export class TatorLocalizationType {
    static BOX = 48;
    static DOT = 49;
    static SUB_BOX = 794;
    static SUB_DOT = 795;

    static isBox(localizationType) {
        return localizationType === this.BOX || localizationType === this.SUB_BOX;
    }

    static isDot(localizationType) {
        return localizationType === this.DOT || localizationType === this.SUB_DOT;
    }
}
